### PR TITLE
Improve docs agent config limits

### PR DESCRIPTION
**Pull request by Soromeister**
### PR Checklist

* [ ] updated test coverage
* [x] external facing docs updated
* [ ] appropriate backport labels added
* [x] not a security concern

### CODE REVIEW

File: website/content/docs/agent/config/config-files.mdx [hunk 1]
```diff
@@ -542,7 +542,7 @@ Refer to the [formatting specification](https://golang.org/pkg/time/#ParseDurati
 
 - `license_path` <EnterpriseAlert inline /> This specifies the path to a file that contains the Consul Enterprise license. Alternatively the license may also be specified in either the `CONSUL_LICENSE` or `CONSUL_LICENSE_PATH` environment variables. See the [licensing documentation](/consul/docs/enterprise/license/overview) for more information about Consul Enterprise license management. Added in versions 1.10.0, 1.9.7 and 1.8.13. Prior to version 1.10.0 the value may be set for all agents to facilitate forwards compatibility with 1.10 but will only actually be used by client agents.
 
-- `limits`: This block specifies various types of limits that the Consul server agent enforces.
+- `limits`: This block specifies various types of limits that the Consul server agent enforces. Declaration differs between Consul Agent Server and Consul Agent Client. For Server, use `limits { ... }` and for Client, use `limits = { ... }`. Not adding the `=` sign for the stanza in the Client config could show the error `expected a map, got slice` upon reloading or restarting the service.
```
Comment (trujillo-adam) [inline website/content/docs/agent/config/config-files.mdx:545]:
```suggestion
- `limits`: This block specifies various types of limits that the Consul server agent enforces. When specifying limits for Consul server agents, use `limits { ... }`. When specifying limits for Consul client agents, use `limits = { ... }`. If the `=` character is excluded from client configurations, Consul prints the following error message when reloading or restarting the service: `expected a map, got slice` .
```
A few suggestions for consistency with our style guide. Also, I think the `limits` block is only intended to be specified in server agent configurations and not in clients.  I need to check with the engineering team, but we may need to change this.

File: website/content/docs/agent/config/config-files.mdx [hunk 2]
```diff
@@ -561,6 +561,8 @@ Refer to the [formatting specification](https://golang.org/pkg/time/#ParseDurati
...
   - `txn_max_req_len` - **(Advanced)** Configures the maximum number of bytes for a transaction request body to the [`/v1/txn`](/consul/api-docs/txn) endpoint. This limit defaults to [raft's](https://github.com/hashicorp/raft) suggested max size (512KB). **Note that tuning these improperly can cause Consul to fail in unexpected ways**, it may potentially affect leadership stability and prevent timely heartbeat signals by increasing RPC IO duration.
 
+  ~> **Warning:** Both `kv_max_value_size` and `txn_max_req_len` values need to be declared in bytes, e.g. for `512KB` the value would be `524288`.
```
Comment (trujillo-adam) [inline website/content/docs/agent/config/config-files.mdx:564]:
```suggestion
  The `kv_max_value_size` and `txn_max_req_len` fields parse values as bytes. For example, specify `524288` to configure a limit of `512KB`.
```

**Review by trujillo-adam — COMMENTED** [review]
Thanks for adding this information. As I noted in my suggestion, I need to verify the use case with the engineering team.

File: website/content/docs/agent/config/config-files.mdx (hunk 1 above)
Comment (trujillo-adam) [reply website/content/docs/agent/config/config-files.mdx:545]:
```suggestion
- `limits`: This block specifies various types of limits that the Consul server agent enforces. Configuring limits does not affect the behavior of Consul agents in client mode. 
```
I verified with the product team that limit configurations do not affect clients. This suggestion makes it explicit that you should only configure these settings for server.

**Comment by trujillo-adam** [comment]
@Soromeister - thanks for bringing this to our attention. I consulted with our product team and they verified that you should only specify the `limits` configurations for servers. I replaced my previous suggestion with a new suggestion that explicitly calls this out. We will approve this PR after you've had a chance to address the feedback. Thanks!

File: website/content/docs/agent/config/config-files.mdx (hunk 1 above)
Comment (Soromeister) [reply website/content/docs/agent/config/config-files.mdx]: Added the suggested change with [this commit](https://github.com/hashicorp/consul/pull/19805/commits/df66f4150f09310dec60487727dac9f8fa81d917).

File: website/content/docs/agent/config/config-files.mdx (hunk 2 above)
Comment (Soromeister) [reply website/content/docs/agent/config/config-files.mdx]: Added the suggested change with [this commit](https://github.com/hashicorp/consul/pull/19805/commits/df66f4150f09310dec60487727dac9f8fa81d917).

**Pull request closed without merging**